sdk/turing/router/config/log_config.py

```diff
@@ -0,0 +1,195 @@
+import re
+import turing.generated.models
+from enum import Enum
+from typing import Optional, Dict, Union
+from turing.generated.model_utils import OpenApiModel
+from turing.router.config.common.schemas import BigQueryTableSchema, KafkaBrokersSchema, KafkaTopicSchema
+
+
+class ResultLoggerType(Enum):
+    NOP = "nop"
+    CONSOLE = "console"
+    BIGQUERY = "bigquery"
+    KAFKA = "kafka"
+
+    def to_open_api(self) -> OpenApiModel:
+        return turing.generated.models.ResultLoggerType(self.value)
+
+
+class LogConfig:
```

Comment:
Hmm I was actually intending for the user the use the child classes (of `LogConfig`), `BigQueryLogConfig` and `KafkaLogConfig` for the time being, to instantiate `LogConfig` objects. These classes only take in the params relative to `bigquery_config` or `kafka_config` respectively and leave the other as None. 

But yeah I'll come up with an additional safeguard to verify the concern you mentioned, so that the user doesn't get away with manipulating the config attributes of `LogConfig` without having the `result_logger_type` set.